File: C++/function.cpp
#include "function.h"

double f(double x){
    return pow(x,4)-x-10;
}

//pow(x,2) - (3 * x) - 4

double df(double x){
    return (4*pow(x,3)) - 1;
}

double ddf(double x){
    return 12*pow(x,2);
}

double g(double x){
    return pow(x+10,0.25);
}

File: Python/function.py
import math

def f(x):
    first = 1/math.sqrt((1.5175**2)-1)
    firstBig = first*(1.5175-x)
    return math.sqrt(1-(x**2)) - firstBig

def df(x): return 0.876113 - (x/math.sqrt(1-(x**2)))

def ddf(x): return math.exp(x)

def g(x): return math.sqrt(1-(x**2))
